mixer/pkg/attribute/envoyProtoBag.go

```diff
@@ -0,0 +1,316 @@
+		reqMap["request.headers"] = attr.WrapStringMap(request.GetRequestHeaders())
+		reqMap["request.host"] = request.GetAuthority()
+		reqMap["request.method"] = request.GetRequestMethod().String()
+		reqMap["request.path"] = request.GetPath()
+		reqMap["request.scheme"] = request.GetScheme()
+		reqMap["request.useragent"] = request.GetUserAgent()
+		reqMap["response.headers"] = attr.WrapStringMap(response.GetResponseHeaders())
+		reqMap["response.code"] = int64(response.GetResponseCode().GetValue())
+		reqMap["response.size"] = int64(response.GetResponseBodyBytes())
+		reqMap["response.total_size"] = int64(response.GetResponseBodyBytes()) + int64(response.GetResponseHeadersBytes())
+		fillContextProtocol(reqMap, request.GetRequestHeaders())
+	} else if tcpLogs := msg.GetTcpLogs(); tcpLogs != nil {
+		commonproperties = *tcpLogs.GetLogEntry()[num].GetCommonProperties()
+		connection := tcpLogs.GetLogEntry()[num].GetConnectionProperties()
+
+		reqMap["context.time"] = reformatTime(commonproperties.GetStartTime(), 0)
+		reqMap["connection.received.bytes"] = int64(connection.GetReceivedBytes())
+		reqMap["connection.sent.bytes"] = int64(connection.GetSentBytes())
+		reqMap["context.protocol"] = "tcp"
+	}
+	reqMap["context.reporter.kind"] = "inbound"
+	fillAddress(reqMap, commonproperties.GetDownstreamLocalAddress(), "destination")
+	fillAddress(reqMap, commonproperties.GetDownstreamRemoteAddress(), "source")
+	tlsproperties := commonproperties.GetTlsProperties()
+	reqMap["destination.principal"] = tlsproperties.GetLocalCertificateProperties().GetSubjectAltName()[0].GetUri()
+	reqMap["source.principal"] = tlsproperties.GetPeerCertificateProperties().GetSubjectAltName()[0].GetUri()
+	reqMap["connection.requested_server_name"] = commonproperties.GetTlsProperties().GetTlsSniHostname()
+	reqMap["context.proxy_error_code"] = ParseEnvoyResponseFlags(commonproperties.GetResponseFlags())
+	pb.upstreamCluster = commonproperties.GetUpstreamCluster()
+
+	pb.reqMap = reqMap
+	scope.Debugf("Returning bag with attributes:\n%v", pb)
+	return pb
+}
+
+//fills in destination.service.name and destination.service.host after the initial bag has been built
+func (pb *EnvoyProtoBag) AddNamespaceDependentAttributes(destinationNamespace string) {
+	parts := strings.Split(pb.upstreamCluster, "|")
+	var host string
+	if len(parts) == 4 {
+		host = parts[3]
+	} else {
+		host = pb.reqMap["request.host"].(string)
+	}
+	pb.reqMap["destination.service.host"] = host
+	namePos := strings.IndexAny(host, ".:")
+	if namePos == -1 {
+		pb.reqMap["destination.service.name"] = host
+		return
+	} else if host[namePos] == ':' {
+		pb.reqMap["destination.service.name"] = host[0:namePos]
+		return
+	}
+	namespacePos := strings.IndexAny(host[namePos+1:], ".:")
+	var serviceNamespace string
+	if namespacePos == -1 {
+		serviceNamespace = host[namePos:]
+	} else {
+		serviceNamespace = host[namePos+1 : namespacePos+namePos+1]
+	}
+	if serviceNamespace == destinationNamespace {
```

Comment:
question: if we know the service namespace, should we be setting `destination.service.namespace` here too?

Reply:
I added it to the last line of the function on 190